components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/IdentityProviderManagementServiceTest.java

```diff
@@ -0,0 +1,1097 @@
+        rm1New.setRemoteRole("Role1New");
+        rm1New.setLocalRole(new LocalRole("1", "LocalRole1"));
+        RoleMapping rm2New = new RoleMapping();
+        rm2New.setRemoteRole("Role2New");
+        rm2New.setLocalRole(new LocalRole("2", "LocalRole2"));
+
+        PermissionsAndRoleConfig prcNew = new PermissionsAndRoleConfig();
+        prcNew.setIdpRoles(new String[]{"Role1New", "Role2New"});
+        prcNew.setRoleMappings(new RoleMapping[]{rm1New, rm2New});
+        idp1New.setPermissionAndRoleConfig(prcNew);
+
+        FederatedAuthenticatorConfig facNew = new FederatedAuthenticatorConfig();
+        facNew.setDisplayName("DisplayName1New");
+        facNew.setName("Name");
+        facNew.setEnabled(true);
+        Property property1New = new Property();
+        property1New.setName("Property1New");
+        property1New.setValue("value1New");
+        property1New.setConfidential(false);
+        Property property2New = new Property();
+        property2New.setName("Property2New");
+        property2New.setValue("value2New");
+        property2New.setConfidential(false);
+        facNew.setProperties(new Property[]{property1New, property2New});
+        idp1New.setFederatedAuthenticatorConfigs(new FederatedAuthenticatorConfig[]{facNew});
+
+        ProvisioningConnectorConfig pcc1New = new ProvisioningConnectorConfig();
+        pcc1New.setName("ProvisiningConfig1");
+        pcc1New.setProvisioningProperties(new Property[]{property1New});
+        ProvisioningConnectorConfig pcc2New = new ProvisioningConnectorConfig();
+        pcc2New.setName("ProvisiningConfig2");
+        pcc2New.setProvisioningProperties(new Property[]{property2New});
+        pcc2New.setEnabled(true);
+        pcc2New.setBlocking(true);
+        idp1New.setProvisioningConnectorConfigs(new ProvisioningConnectorConfig[]{pcc1New, pcc2New});
+
+        ClaimConfig claimConfigNew = new ClaimConfig();
+        claimConfigNew.setLocalClaimDialect(false);
+        claimConfigNew.setRoleClaimURI("Country");
+        claimConfigNew.setUserClaimURI("Country");
+        ClaimMapping cm = ClaimMapping.build("http://wso2.org/claims/country", "Country", "", true);
+        Claim remoteClaim = new Claim();
+        remoteClaim.setClaimId(0);
+        remoteClaim.setClaimUri("Country");
+        claimConfigNew.setClaimMappings(new ClaimMapping[]{cm});
+        claimConfigNew.setIdpClaims(new Claim[]{remoteClaim});
+        idp1New.setClaimConfig(claimConfigNew);
+
+        // Initialize New Test Identity Provider 2.
+        IdentityProvider idp2New = new IdentityProvider();
+        idp2New.setIdentityProviderName("testIdP2New");
+
+        // Initialize New Test Identity Provider 3.
+        IdentityProvider idp3New = new IdentityProvider();
+        idp3New.setIdentityProviderName("testIdP3New");
+
+        return new Object[][]{
+                {"testIdP1", idp1New},
+                {"testIdP2", idp2New},
+                {"testIdP3", idp3New},
```

Comment:
better if we can mention the differences in these test IDPs.

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/3493/commits/ef44da16e93a4a038f814603c42eadfb3a300f15